distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestCatalogValidation.java

```diff
@@ -1065,21 +920,85 @@ private ValidatableResponse executeOpenSearch(String format, String... query) {
                 .then();
     }
 
-    protected String ingestXmlFromResource(String resourceName) throws IOException {
+    protected String ingestXmlFromResource(String resourceName, boolean shouldFail)
+            throws IOException {
         StringWriter writer = new StringWriter();
         IOUtils.copy(IOUtils.toInputStream(getFileContent(resourceName)), writer);
-        return ingest(writer.toString(), "text/xml");
-    }
-
-    protected String ingestXmlFromResource(String resourceName, boolean checkResponse)
-            throws IOException {
-        if (checkResponse) {
-            return ingestXmlFromResource(resourceName);
+        List<String> ids = new ArrayList<>();
+        String id;
+        expect("Ingest failed to meet expectations").within(10, TimeUnit.SECONDS)
+                .checkEvery(1, TimeUnit.SECONDS)
+                .until(() -> {
+                    try {
+                        ids.add(ingest(writer.toString(), "text/xml", true));
+                        return !shouldFail;
+                    } catch (Error e) {
```

Comment:
❗️ Maybe we shouldn't re-use the `ingest` method here and have a different, common method that does an ingest with retry? If not, we may want to specifically catch `AssertionError` instead of `Error`.